enhancements/subscription-content/subscription-content-access.md

```diff
@@ -0,0 +1,312 @@
+
+#### Access Appropriate RHEL Content for Container
+
+As a developer using OpenShift I want to be able to access appropriate RHEL content for my container
+So that I download RHEL 8 content in UBI 8 containers And download RHEL 7 content in UBI 7
+containers
+
+#### Provide Subscriptions to Multiple Clusters
+
+As a multi-cluster administrator, I want to enable my cluster for entitlements via a declarative API
+that works well with GitOps (or similar patterns) so that granting access to subscription content is
+not a burden as I scale the number of clusters under management.
+
+### Implementation Details/Notes/Constraints [optional]
+
+To consume RHEL content in OpenShift, the running container must have access to the following:
+
+One or more entitlement keys, which are PEM-encoded certificates. A yum repository definition,
+which tells yum/dnf where to access content from.
+
+Yum repository definitions are specific to the major version of RHEL that the running container is
+based on - a RHEL 8-based container is not compatible with RHEL 7 content.
+
+This enhancement proposal will detail at a high level the functions of:
+
+1. A cluster-wide subscription manager to generate subscription content access data
+2. A projected resource CSI driver to share resources across OpenShift namespaces
+3. A subscription content webhook to generate the necessary volume mounts to add subscription
+   content access data to containers
+4. Extensions to propagate subscription content access data into OpenShift builds.
+
+#### Cluster Subscription Manager
+
+The cluster subscription manager will take as input the `cloud.redhat.com` pull secret, which
+already provides authentication for the cluster and provides the means to register clusters with
+Red Hat’s cloud manager. The pull secret is then used to obtain the following:
+
+1. An entitlement key associated with the subscription for the OpenShift cluster.
+2. For Satellite clusters, the `redhat.repo` yum repo configurations for all supported major
+   versions of RHEL. Today these will be RHEL 7 and 8 - future major versions of RHEL must be
+   anticipated.
+
+Like RHSM, the cluster subscription manager obtains the entitlement keys associated with the
+attached OpenShift subscription. It is assumed that OpenShift entitlements grant access to RHEL
+content, but may not grant access to layered content such as EAP or Application Runtimes. As an
+alternative, the subscription manager may try to obtain a Simple Content Access certificate which
+globally grants access to yum repos available to the customer account. This content must be saved
+in a Secret within the `openshift-config` namespace, named `etc-pki-entitlement`.
+
+The cluster subscription manager must also manage entitlement key rotation events. In the event a
+cluster's entitlement key is changed, the cluster subscription manager must update the key stored
+in `openshift-config/etc-pki-entitlement`.
```

Comment:
Updated to include certificate rotation. Entitlement keys can be revoked and rotated.

Reply:
How does the cluster subscription manager *receive* the key rotation events @adambkaplan ?  Is it the thing that registers the cluster with Red Hat's cloud manager (and the registration includes supplying some sort of callback/listener)?  Or does some other component within OCP already doing registering and recording events somewhere?

Some hint on the mechanics could be useful here IMO.

Reply:
@gabemontero to be honest, I don't know enough w.r.t entitlement rotation/revocation works today to say how this would be implemented. I can add this as an open question - presumably there are existing APIs/infra to do this in RHEL today.